### ISSUE TITLE
Perlin Lib not compatible with MUD2

### ISSUE THREAD

**Issue by bitzoic** [latticexyz/mud#935]
#### Description

Currently, MUD2 does not play nicely with public libraries. The functions in the [Perlin.sol](https://github.com/latticexyz/mud/blob/main/packages/noise/contracts/Perlin.sol) file all have `public` modifiers. 

#### Motivation

Inability to use Perlin noise in systems.

#### Minimal Repro

```solidity
// SPDX-License-Identifier: MIT
pragma solidity >=0.8.0;

import { System } from "@latticexyz/world/src/System.sol";
import { Perlin } from "@latticexyz/noise/contracts/Perlin.sol";

contract MySystem is System {

    function callsPerlin(int256 positionX, int256 positionY) public pure returns (uint256) {
        int128 perlinNoise = Perlin.noise2d(positionX, positionY, 0, 0);
    }
}
```

**Comment by alvrs** [comment]
fixed in #934, you can update your project to the latest MUD with `pnpm mud:up` (assuming you started from a template)

**Comment by bitzoic** [comment]
Thanks!
